test/complex.jl

```diff
@@ -15,6 +15,13 @@ end
 @test sprint(show, complex(1, 0), context=:compact => true) == "1+0im"
 @test sprint(show, complex(true, true)) == "Complex(true,true)"
 
+@testset "unary operator on complex boolean" begin
+    @test isequal(+(complex(true)), complex(1, 0))
+    @test isequal(+(complex(false)), complex(0, 0))
+    @test isequal(-(complex(true)), complex(-1, 0))
+    @test isequal(-(complex(false)), complex(0, 0))
```

Comment:
Maybe test all four complex bool values? Or at least one which has a non-zero imaginary part.

Reply:
well complex(bool) is always giving a zero imaginary part, i haven't tested the other way round

Reply:
> four complex bool values
What exactly do you mean by this?

Reply:
There are four `Complex{Bool}` values:
1. `Complex(false, false)`
2. `Complex(false, true)`
3. `Complex(true, false)`
4. `Complex(true, true)`

Reply:
Ok, cool. I'm adding these tests!

Reply:
Right, the other thing to do would be to also test the application of `+` and `-` to those four values.

Reply:
Now I think that the tests are both comprehensive and exhaustive